liferay-echo/liferay-portal
# LPS-114186 Migrate AUI tree-view in asset-taglib to React

Hey @liferay-echo  ,

This PR is follow up on

julien#284
julien#285
mateomustapic#9
mateomustapic#10
https://github.com/wincent/liferay-portal/pull/311
https://github.com/wincent/liferay-portal/pull/313
and
https://github.com/julien/liferay-portal/pull/286

This PR is a work on migration of `AUI tree-view` to `React` in `asset-taglib` module.
The old aui tree-view implementation had completely different logic than React TreeView requires, so Eudaldo created new `Java` classes which can now be used in new implementation.

I used the `<react:component>` tag and included React component in the jsp where old tree-view was located, and I additionally cleaned up unnecessary jsp and AUI code which was used in old implementation.
Additionally, Julien made modifications to react component for passing `categories` and `vocabularies` to `AssetCategoriesNavigationTreeView` component.

After Eudaldo's comments in https://github.com/wincent/liferay-portal/pull/313
I have made the required `JS` changes he requested.
The new component now behaves as the old one, please @ealonso confirm if this is true.

To test these changes :

Navigate to Categorization / Categories
Create some categories in Vocabularies list for test
Create a widget page (Site Builder/Pages new Blank Page)
Add the portlet categories navigation to a page
Now the tree view should be visible on created page.
